### ISSUE TITLE
Moving user to its own schema (i.e on org creation) leaves triggers on public schema

### ISSUE THREAD

**Issue by zenitraM** [CartoDB/cartodb#6295]
Steps to replicate:
- Create an user, import some table
- Add a trigger or materialized view using SQL API. It will be created on the "public" schema.
- Add an organization to the user. 

This causes all tables to be moved to the "username" schema from the "public" schema, but will leave triggers / matviews / things behind. Those will get discarded by the user mover, as it moves schemas one by one and discards everything on the "public" one,

**Comment by juanignaciosl** [comment]
@zenitraM I've done [a test about triggers](https://github.com/CartoDB/cartodb/compare/6295-Moving_user_to_its_own_schema_leaves_triggers_on_public_schema?expand=1) and test passes. Have you been able to reproduce it with triggers?

Next step is checking materialized views.

**Comment by zenitraM** [comment]
@juanignaciosl hm, I will try to reproduce locally, but it looks like the moving method only explicitly ALTERs the tables, nothing else:
https://github.com/CartoDB/cartodb/blob/master/app/models/user/db_service.rb#L276

so I'm not sure what could be moving the triggers if they are getting moved :/

**Comment by juanignaciosl** [comment]
PS: materialized views moving test is uploaded, and it fails, so I guess the problems are related to that.

**Comment by juanignaciosl** [comment]
Well, since triggers are attached to tables it'd make sense that moving the table moved the triggers, but I can't find information about that in PostgreSQL documentation, I'll keep searching :eyes:

**Comment by juanignaciosl** [comment]
In fact (taken from [here](http://www.postgresql.org/docs/9.1/static/sql-createtrigger.html): "the trigger inherits the schema of its table" might explain that. I'll fix materialized views anyway, can you tell me other DB objects that you've found issues with?

**Comment by zenitraM** [comment]
Hmm, not sure what could have left the trigger then :sweat_smile: . The broken case was moving this:

```
ERROR:  function public.update_updated_at() does not exist
```

Maybe the correct way to do this is renaming the "public" schema instead of moving the objects one by one... although not sure if that might bring some other implications :/

**Comment by juanignaciosl** [comment]
Could _functions_ be the problem, instead of _triggers_?

**Comment by azamorano** [comment]
There is an example of missing relation too:
```
ERROR:  relation "public.seq_cartodb_processed_3078476315368391601" does not exist
```

**Comment by zenitraM** [comment]
Oh right, it could be the _functions_ that are left. I was mixing both.

**Comment by juanignaciosl** [comment]
Ok, this will fix #5579 as well.
